website/app/components/doc/page/header/search/index.js

```diff
@@ -0,0 +1,204 @@
+  @tracked isModalOpen = false;
+
+  @action
+  didInsertSearchContainer(element) {
+    const autocompleteInstance = autocomplete({
+      id: 'doc-algolia-search-autocomplete-container', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-id
+      container: element, // 'if necessary you can use an ID eg. `#doc-algolia-search-autocomplete-container',` // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-container
+      // panelContainer: '...' // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-panelcontainer
+      panelPlacement: 'full-width', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-panelplacement
+      placeholder: 'Search', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-placeholder
+      // classNames: {}, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-classnames
+      // plugins: [algoliaRecentSearchesPlugin], // see: https://www.algolia.com/doc/ui-libraries/autocomplete/guides/adding-recent-searches/
+      openOnFocus: true, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-openonfocus
+      // debug: true, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-debug
+      detachedMediaQuery: '', // https://www.algolia.com/doc/ui-libraries/autocomplete/core-concepts/detached-mode/
+      translations: {
+        detachedCancelButtonText: 'Close', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-translations
+      },
+      onStateChange: ({ state, prevState }) => {
+        // used to reset the query to an empty state/string when the modal is closed (via "click" or "esc" key)
+        if (!state.isOpen && prevState.isOpen) {
+          autocompleteInstance.setQuery('');
+        }
+      },
+
+      //
+      // GET SOURCES
+      // see: https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-getsources
+      getSources({ query }) {
+        if (query.trim() !== '') {
+          return [
+            {
+              sourceId: 'content',
+              getItems: getItemsFunction({
+                searchQuery: query,
+                searchParams: {
+                  hitsPerPage: 5,
+                  attributesToHighlight: ['pageTitle', 'content'],
+                  attributesToSnippet: ['content:20'],
+                  distinct: true,
+                  // attributeForDistinct: 'pageBaseURL', // this doesn't work, for some reason; it needs to be set at Algolia dashboard level under "Deduplication and Grouping" configuration
+                },
```

Comment:
In the documentation I found it says `attributeForDistinct` can be used in the `setSettings` method so maybe it's not supposed to work here? 

https://www.algolia.com/doc/api-reference/api-parameters/attributeForDistinct/

Reply:
@KristinLBradley We're using the `getAlgoliaResults` function (docs [here](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/getAlgoliaResults/)) which points to the [Search API Parameters](https://www.algolia.com/doc/api-reference/search-api-parameters/] which exposes the `[distinct](https://www.algolia.com/doc/api-reference/api-parameters/distinct/)` parameter, but strangely enough not the `attributeForDistinct` (strangely because they both appear together in the Algolia dashboard settings)
<img width="1468" alt="screenshot_3298" src="https://github.com/hashicorp/design-system/assets/686239/73163dac-5e77-487c-812f-b378feb10532">